packages/frontend/src/user/login.tsx

```diff
@@ -30,6 +30,10 @@ type EmailAndPassword = {
 export function Login(props: {
     onComplete?: (user: User) => void;
 }) {
+    const [userAdded, setUserAdded] = createSignal<
+        "Please complete login" | "Please complete signup" | ""
```

Comment:
This should become irrelevant in my proposed approach, but for the sake of learning, I should point out that there is hardly ever a situation where you want to elevate human-readable text like this into the type level. This requires that the message be repeated exactly, which is itself generally a code smell.